sklearn/manifold/tests/test_t_sne.py

```diff
@@ -640,22 +640,23 @@ def test_barnes_hut_angle():
 def test_n_iter_without_progress():
     # Use a dummy negative n_iter_without_progress and check output on stdout
     random_state = check_random_state(0)
-    X = random_state.randn(100, 2)
-    tsne = TSNE(n_iter_without_progress=-1, verbose=2, learning_rate=1e8,
-                random_state=1, method='exact', n_iter=300)
-
-    old_stdout = sys.stdout
-    sys.stdout = StringIO()
-    try:
-        tsne.fit_transform(X)
-    finally:
-        out = sys.stdout.getvalue()
-        sys.stdout.close()
-        sys.stdout = old_stdout
-
-    # The output needs to contain the value of n_iter_without_progress
-    assert_in("did not make any progress during the "
-              "last -1 episodes. Finished.", out)
+    X = random_state.randn(100, 10)
+    for method in ["barnes_hut", "exact"]:
+        tsne = TSNE(n_iter_without_progress=-1, verbose=2, learning_rate=1e8,
+                    random_state=1, method='barnes_hut', n_iter=351)
```

Comment:
I changed it in the last commit. But I am not able to fix the test with `method=exact`. I did not find any good example which triggered the stopping criterion. Should I test it only on `method=barnes_hut`? Note that this test is not super stable as changing the parameters might cause failure even with barnes hut. It was previously stable because the stopping criterion could be triggered with `early_exaggeration`.